python/tank/util/metrics.py

```diff
@@ -345,79 +348,164 @@ def data(self):
-        
-        :param str attr_name: Name of the attribute.
-        :param str attr_value: The value of the attribute.
+        # TODO: find a descent library and implement method
+        pass
 
+    def add_system_info_properties(self):
         """
-        super(UserAttributeMetric, self).__init__({
-            "type": "user_attribute",
-            "attr_name": attr_name,
-            "attr_value": attr_value,
-        })
+        Helper method adding a number of system information metrics such as:
+        - OS type
+        - OS version
+        - System architechture
+        - Manufacturer etc.
+
+        The method takes care of gathering relevant information and properly
+        formatting them for submission to a Shotgun site
+        """
+        # TODO: find a descent library and implement method
+        pass
 
     def __repr__(self):
-        """Official str representation of the user attribute metric."""
-        return "%s.%s" % (self._data["attr_name"], self._data["attr_value"])
+        """Official str representation of the user activity metric."""
+        return "%s:%s" % (self._data["event_group"], self._data["event_name"])
 
 
 ###############################################################################
 # metrics logging convenience functions
 
-def log_metric(metric, log_once=False):
-    """Log a Toolkit metric.
-    
-    :param ToolkitMetric metric: The metric to log.
-    :param bool log_once: ``True`` if this metric should be ignored if it has
-        already been logged. Defaults to ``False``.
+def log_event_metric(metric_event, log_once=False):
+    """ Log a Toolkit metric event now using the Amplitude service.
 
-    This method simply adds the metric to the dispatch queue.
-    """
-    MetricsQueueSingleton().log(metric, log_once=log_once)
+    This method adds the metric event to a dispatch queue meaning that the
+    metric doesn't get posted on the web right away. The can add a few
+    metrics together in a single payload. The dispatcher processes the
+    queue every 5-15 seconds (subject to change).
 
-def log_user_activity_metric(module, action, log_once=False):
-    """Convenience method for logging a user activity metric.
+    :param EventMetric metric: A metric event structure to add the the queue.
 
-    :param str module: The module the activity occured in.
-    :param str action: The action the user performed.
     :param bool log_once: ``True`` if this metric should be ignored if it has
         already been logged. Defaults to ``False``.
+
     """
-    log_metric(UserActivityMetric(module, action), log_once=log_once)
+    MetricsQueueSingleton().log(metric_event, log_once=log_once)
 
-def log_user_attribute_metric(attr_name, attr_value, log_once=False):
-    """Convenience method for logging a user attribute metric.
+def log_metric(metric, log_once=False):
+    """ Depricated method, use the `log_metric_event` method."""
```

Comment:
If we go with `EventMetric.log`, then just a reminder to update these docstrings.